debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotChangeEventSource.java

```diff
@@ -616,4 +617,14 @@ protected Table refreshTableSchema(Table table) throws SQLException {
     private KeyMapper getKeyMapper() {
         return connectorConfig.getKeyMapper() == null ? table -> table.primaryKeyColumns() : connectorConfig.getKeyMapper();
     }
+
+    private List<Column> getQueryColumns(Table table) {
+        if (connectorConfig.getIncrementalSnapshotChunkKeyColumns() != null){
+            String cfg  = connectorConfig.getIncrementalSnapshotChunkKeyColumns();
+            return CustomKeyMapper.getInstance(cfg, x -> x.schema() + "." + x.table()).getKeyKolumns(table);
```

Comment:
The table mapper is specific to PostgreSQL, still thinking how to get this one to be database agnostic.

Reply:
Sorry but maybe I haven't had enough coffee, could you explain?

Reply:
The `getInstance` method needs an implementation for `TableIdToStringMapper` interface. I've noticed that for SQL Server and PG it's always used `x -> x.schema() + "." + x.table()`. Since this is an abstract class, I was thinking that this interface implementation should be sent when this class is being initialized maybe and not "hardcoded" since it might not apply for all implementations.

Reply:
No need for that, simply have a method that returns the lambda that most connectors use and then override that method for connectors that need a different implementation as we've done with other methods such as `preReadChunk` or `init` to name a few.

Reply:
Just to make sure we are on the same page. If both key mapping and surrogate incremental snapshot column is used the the surrogate column is used for generating the SQL statements but the key mapper will be used to create keys in data messages, right?

Reply:
So, the usage here of `CustomKeyMapper` is a bit of cheating from my side, I needed all the logic from [here](https://github.com/debezium/debezium/blob/main/debezium-core/src/main/java/io/debezium/relational/Key.java#L112) in order to get the list of `Column` objects, from a string config, that would be used to build the SQL statements. Probably I should just create a new method for the same logic just to not be that confusing? This code does not change how the key mapper is used to create keys in data messages.